### PR TITLE
Removing ui

### PR DESCRIPTION
**Pull request by aanm**

### CODE REVIEW

File: daemon/daemon/labels.go [hunk 1]
```diff
@@ -95,8 +95,6 @@ func (d *Daemon) PutLabels(lbls labels.Labels, contID string) (*policy.Identity,
 
 	log.Debugf("Incrementing label %d ref-count to %d\n", secCtxLbls.ID, secCtxLbls.RefCount())
 
-	d.AddOrUpdateUINode(secCtxLbls.ID.Uint32(), secCtxLbls.Labels.ToSlice(), secCtxLbls.RefCount())
```
Comment (aanm) [inline daemon/daemon/labels.go:98]: @tgraf here

File: daemon/daemon/labels.go [hunk 2]
```diff
@@ -208,12 +206,6 @@ func (d *Daemon) DeleteLabelsBySHA256(sha256Sum string, contID string) error {
 		return err
 	}
 
-	if dbSecCtxLbls.RefCount() == 0 {
-		d.DeleteUINode(dbSecCtxLbls.ID.Uint32())
```
Comment (aanm) [inline daemon/daemon/labels.go:212]: @tgraf here

File: daemon/daemon/labels.go [hunk 3]
```diff
@@ -208,12 +206,6 @@ func (d *Daemon) DeleteLabelsBySHA256(sha256Sum string, contID string) error {
...
-	} else {
-		d.AddOrUpdateUINode(dbSecCtxLbls.ID.Uint32(), dbSecCtxLbls.Labels.ToSlice(), dbSecCtxLbls.RefCount())
```
Comment (aanm) [inline daemon/daemon/labels.go:214]: @tgraf here

File: daemon/daemon/state.go [hunk 4]
```diff
@@ -107,9 +107,6 @@ func (d *Daemon) SyncState(dir string, clean bool) error {
 		if err != nil {
 			log.Warningf("Failed while updating ep %d: %s", ep.ID, err)
 		} else {
-			if ep.SecLabel != nil {
-				d.AddOrUpdateUINode(ep.SecLabel.ID.Uint32(), ep.SecLabel.Labels.ToSlice(), ep.SecLabel.RefCount())
```
Comment (aanm) [inline daemon/daemon/state.go:111]: @tgraf here

**Review by aanm — COMMENTED** [review]
The places where I wrote "here" are the ones where the event is triggered to show up in the web UI.

**Comment by tgraf** [comment]
> The places where I wrote "here" are the ones where the event is triggered to show up in the web UI.

We should define an interface{} which contains hooks that can later be implemented to trigger sending events to an aggerator. Initially the default implementation of the interface{} can be a no-operation.

File: plugins/cilium-docker/driver/driver.go [hunk 5]
```diff
@@ -61,7 +61,7 @@ func NewDriver(ctx *cli.Context) (Driver, error) {
 		log.Fatalf("Error while starting cilium-client: %s", err)
 	}
 
-	for tries := 0; tries < 10; tries++ {
+	for tries := 0; tries < 20; tries++ {
```
Comment (tgraf) [inline plugins/cilium-docker/driver/driver.go]: Just bump this to 5 minutes or indefinitely.

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
>We should define an interface{} which contains hooks that can later be implemented to trigger sending events to an aggerator. Initially the default implementation of the interface{} can be a no-operation.

What about replacing the `here` lines with a go channel (`d.UIEvent<-Event{Type: ADD, Obj: Interface{}}`)?

**Comment by tgraf** [comment]
> What about replacing the here lines with a go channel (d.UIEvent<-Event{Type: ADD, Obj: Interface{}})?

Let's not make it UI specific and call it event queue or something. This will be useful for other things besides a UI.

**Comment by aanm** [comment]
>Let's not make it UI specific and call it event queue or something. This will be useful for other things besides a UI.

Check the b70bdaa commit.

File: daemon/daemon/daemon.go [hunk 6]
```diff
@@ -421,13 +423,16 @@ func NewDaemon(c *Config) (*Daemon, error) {
 		endpointsDockerEP:         make(map[string]*endpoint.Endpoint),
 		endpointsLearning:         make(map[uint16]labels.LearningLabel),
 		endpointsLearningRegister: make(chan labels.LearningLabel, 1),
-		loadBalancer:              lb,
-		consumableCache:           policy.NewConsumableCache(),
-		policy:                    policy.Tree{Root: policy.NewNode(common.GlobalLabelPrefix, nil)},
-		ignoredContainers:         make(map[string]int),
+		events:            make(chan events.Event, 10),
```
Comment (tgraf) [inline daemon/daemon/daemon.go]: Is 10 enough of a buffer? Does it make sense to bump it to 512 as things like policy changes may trigger a lot of events and we would need to start waiting for the readers.

File: daemon/daemon/events.go [hunk 7]
```diff
@@ -0,0 +1,25 @@
+//
+// Copyright 2016 Authors of Cilium
```
Comment (tgraf) [inline daemon/daemon/events.go:2]: We can start adding -2017 for new headers.

**Review by tgraf — APPROVED** [review]

File: daemon/daemon/daemon.go (hunk 6 above)
Comment (aanm) [reply daemon/daemon/daemon.go]: I think the best option is to trigger a new go routine for each reader.
```golang
newEvent := <-...
for listener := range listeners {
   go func(listener chan event) {
        select {
         case listener <- newEvent:
         case <-time.After(1 * time.Second):
               log.Debug("listener foo didn't receive new event")
       }(listener)
}
```

Comment (tgraf) [reply daemon/daemon/daemon.go]: I'm not worried about listeners yet, the `listenForCiliumEvents()` itself needs to be scheduled to run to drain the 10 events before the main daemon continues running.

Comment (aanm) [reply daemon/daemon/daemon.go]: Ah I see, ok I'll change it 512.

**Pull request merged**